src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/TagHelperSemanticRangeVisitor.cs

```diff
@@ -78,10 +78,26 @@ public override void VisitMarkupEndTag(MarkupEndTagSyntax node)
 
         public override void VisitMarkupCommentBlock(MarkupCommentBlockSyntax node)
         {
-            Debug.Assert(node.Children.Count == 3, $"There should be 3 nodes but were {node.Children.Count}");
             AddSemanticRange(node.Children[0], RazorSemanticTokensLegend.MarkupCommentPunctuation);
-            AddSemanticRange(node.Children[1], RazorSemanticTokensLegend.MarkupComment);
-            AddSemanticRange(node.Children[2], RazorSemanticTokensLegend.MarkupCommentPunctuation);
+
+            for(var i = 1; i < node.Children.Count - 1; i++)
+            {
+                var commentNode = node.Children[i];
+                switch(commentNode.Kind)
+                {
+                    case SyntaxKind.MarkupTextLiteral:
+                        AddSemanticRange(commentNode, RazorSemanticTokensLegend.MarkupComment);
+                        break;
+                    case SyntaxKind.CSharpCodeBlock:
+                        // This will be handled by the CSharp LSP
+                        break;
+                    default:
```

Comment:
Can anyone point out other kinds of nodes that could be in this? It never even occurred to me to put C# in an html comment so I can guess that my idea of what's possible is not the whole picture.

Reply:
Well technically the C# block can have more nested HTML inside of it which then allows the entire Razor syntax 😄